builtin/game/item_entity.lua

```diff
@@ -213,9 +213,31 @@ core.register_entity(":__builtin:item", {
 		end
 	end,
 
-	on_punch = function(self, hitter)
-		local inv = hitter:get_inventory()
-		if inv and self.itemstring ~= "" then
+	on_punch = function(self, hitter, ...)
+		if self.itemstring ~= "" then
+			-- Call the on_pickup callback of item definition.
+			local item_def = core.registered_items[self.itemstring]
+			if item_def and item_def.on_pickup and
+					not item_def.on_pickup(self, hitter, ...) then
+				return
+			end
+
+			-- Call every function registered on pickup.
```

Comment:
`-- Invoke all on_pickup callbacks`
might probably sound better.

Reply:
Well, it's just a code comment.